pkg/cached/cache.go

```diff
@@ -209,23 +226,21 @@ func NewTransformer[T, V any](transformerFn func(Result[T]) Result[V], source Da
 	})
 }
 
-// NewSource creates a new cache that generates some data. This
-// will always be called since we don't know the origin of the data and
-// if it needs to be updated or not.
-func NewSource[T any](sourceFn func() Result[T]) Data[T] {
-	c := source[T](sourceFn)
-	return &c
+// DataFunc creates a new cache that always reruns the function.
+func DataFunc[T any](fn func() Result[T]) Data[T] {
```

Comment:
the constructor is not really needed. Compare 
```golang
// The HandlerFunc type is an adapter to allow the use of
// ordinary functions as HTTP handlers. If f is a function
// with the appropriate signature, HandlerFunc(f) is a
// Handler that calls f.
type HandlerFunc func(ResponseWriter, *Request)

// ServeHTTP calls f(w, r).
func (f HandlerFunc) ServeHTTP(w ResponseWriter, r *Request) {
	f(w, r)
}
```

But it looks like with generics, Golang will derive `T` automatically when calling a func, not for a type-cast.